spec/lib/pre_assembly/batch_spec.rb

```diff
@@ -152,27 +157,78 @@
-      # #size on the Enumerator object is calculated in a way that avoids iterating over the whole thing
-      expect(flat_dir_images.digital_objects.size).to eq(flat_dir_images.digital_objects.to_a.size)
-    end
+      it 'calculates size correctly' do
+        # #size on the Enumerator object is calculated in a way that avoids iterating over the whole thing
+        expect(flat_dir_images.digital_objects.size).to eq(flat_dir_images.digital_objects.to_a.size)
+      end
 
-    it 'finds the correct number of objects' do
-      expect(batch.digital_objects.size).to eq(3)
-    end
+      it 'finds the correct number of objects' do
+        expect(batch.digital_objects.size).to eq(3)
+      end
 
-    it 'handles containers correctly' do
-      expect(multimedia.digital_objects.first.container.size).to be > multimedia.staging_location.size
+      it 'sets the correct relative_paths for the files' do
+        expect(relative_file_paths).to eq [
+          [
+            'image1.tif',
+            'image2.tif'
+          ],
+          [
+            'image1.tif',
+            'image2.tif'
+          ],
+          [
+            'image1.tif',
+            'image2.tif'
+          ]
+        ]
+      end
+
+      it 'handles containers correctly' do
+        expect(multimedia.digital_objects.first.container.size).to be > multimedia.staging_location.size
+      end
+
+      it 'augments the digital objects with additional information' do
+        expect(flat_dir_images.digital_objects.size).to eq(3)
+        extra_info = flat_dir_images.digital_objects.map { |dobj| { label: dobj.label, source_id: dobj.source_id } }
+        expect(extra_info).to eq [
+          {
+            label: 'Avus 1937',
+            source_id: 'foo-1.0'
+          },
+          {
+            label: 'Avus 1938',
+            source_id: 'foo-2.1'
+          },
+          {
+            label: 'Avus 1938, 1956',
+            source_id: 'foo-2.2'
+          }
+        ]
+      end
     end
 
-    it 'augments the digital objects with additional information' do
-      expect(flat_dir_images.digital_objects.size).to eq(3)
-      flat_dir_images.digital_objects.each do |dobj|
-        expect(dobj.label).to be_a(String)
-        expect(dobj.label).not_to eq('Unknown') # hardcoded in class
-        expect(dobj.source_id).to be_a(String)
+    context 'with hierarchical file structure' do
+      let(:batch) { batch_setup(:hierarchical_files) }
+
+      it 'finds the correct number of objects' do
+        expect(batch.digital_objects.size).to eq(1)
+      end
+
+      it 'sets the correct relative_paths for the files' do
+        expect(relative_file_paths).to eq [
+          [
+            'config/settings/test.yml',
+            'config/settings/test1.yml',
+            'config/settings/test2.yml',
+            'config/test.yml',
+            'images/image.jpg',
+            'images/subdir/image.jpg',
+            'test1.txt'
+          ]
+        ]
       end
```

Comment:
some new expectations to ensure we set the relative paths correctly for both flat and hierarchical folder structures